pandas/tests/io/formats/style/test_html.py

```diff
@@ -780,6 +780,8 @@ def test_hiding_index_columns_multiindex_trimming():
         ("ftp scheme: ftp://www.web", True, "ftp://www.web"),
         ("subdirectories: www.web.com/directory", True, "www.web.com/directory"),
         ("Multiple domains: www.1.2.3.4", True, "www.1.2.3.4"),
+        ("with port: http://web.com:80", True, "http://web.com:80"),
+        ("with anchor: http://web.com/anc#hor", True, "http://web.com/anc#hor"),
```

Comment:
add all the special characters to this line so that they are tested simultaneously.

Reply:
Thanks, I made the tests more exhaustive and also tweaked the regex a little bit more.